CMakeLists.txt

```diff
@@ -2,84 +2,66 @@
 cmake_minimum_required (VERSION 2.8)
 project (ConnectSDK)
 
-if (DEFINED IOS_PLATFORM)
-
-    if (NOT DEFINED ENV{CONNECT_PREBUILT_ROOT})
-        message (FATAL_ERROR "Please, define CONNECT_PREBUILT_ROOT environment variable")
-    endif ()
-
-    set (BOOST_IOS_ROOT $ENV{CONNECT_PREBUILT_ROOT}/boost)
-    set (Boost_INCLUDE_DIRS ${BOOST_IOS_ROOT}/include)
-    set (Boost_LIBRARY_DIRS ${BOOST_IOS_ROOT}/lib)
-    set (Boost_LIBRARIES "boost")
-
-    set (CURL_IOS_ROOT $ENV{CONNECT_PREBUILT_ROOT}/curl)
-    set (CURL_INCLUDE_DIRS ${CURL_IOS_ROOT}/include)
-    set (CURL_LIBRARY_DIRS ${CURL_IOS_ROOT}/lib)
-    set (CURL_LIBRARIES "curl")
-
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98 -stdlib=libc++ -isysroot ${CMAKE_OSX_SYSROOT} -miphoneos-version-min=9.0")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -isysroot ${CMAKE_OSX_SYSROOT} -miphoneos-version-min=9.0")
-
-    if (${SIMULATOR})
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mios-simulator-version-min=9.0")
+if (CMAKE_CROSSCOMPILING)
+    if (NOT DEFINED PRISM_PLATFORM)
+        message (FATAL_ERROR "When crosscompiling, PRISM_PLATFORM must be set")
+    endif ()  
+elseif (NOT DEFINED PRISM_PLATFORM)
+    find_program (CMAKE_UNAME uname /bin /usr/bin /usr/local/bin)
+    if (CMAKE_UNAME)
+	    exec_program(uname OUTPUT_VARIABLE PRISM_HOST_OS_NAME)
+	    if (PRISM_HOST_OS_NAME STREQUAL "Darwin")
```

Comment:
If you remember in cpp-engine & PVB, build scripts where shared and were in SVN linked project similar to git submodule. We may need same thing here soon. In this case all this platform selection code shall be there. Because otherwise we will have code duplication. Until root file is small and platform/ is not submodule we can live like it is, but it is not clean solution I think.